output/marshal.go

```diff
@@ -136,14 +136,14 @@ func MarshalEntries(ctx context.Context, output *Output, entries []source.Entry,
 			if err != nil {
 				return nil, errors.Wrap(err, fmt.Sprintf("aliases.%d: evaluating entry alias", idx))
 			}
-			if alias == "" {
+			if alias == nil || *alias == "" {
```

Comment:
I feel like it's worth explaining in a brief comment why we have to do this, but I'm not sure exactly where.

You could call it out in a couple of places separately, like here in `MarshalEntries` and somewhere in `js_eval.go` - just because this may feel like a redundant check if you don't know what's going on

Reply:
Fixing the comment about `BeNil` means that this extra check isn't needed now, because `EvaluateResultType` now returns a pointer, not `""` when you ask for a string but we can't evaluate to a string.